### PR TITLE
Export animations only, fixes for glTF and Maya exporter using MEL

### PR DESCRIPTION
**Pull request by noalak**
This PR is resolving issues:
- #537 (linked to #525) : Export animations only (linked to PR https://github.com/BabylonJS/Babylon.js/pull/7252 for import)
- #599 : Expose Maya exporter to MEL commands (see Maya/MELScripts/exportScriptExemple.mel)
- #626 : glTF default values
- #627 : Duplicate glTF texture objects

Disclaimer:
The 'export animations only' feature is implemented for 3ds Max.
@Drigax, @elpie89, @Selmar are invited to test the export and import process. Samples are provided in the exporter AnimationsOnly samples folder or can be downloaded via [WeTransfer here](https://wetransfer.com/downloads/f25c29812c22cc91b4e3c8376019ffd720191203132002/004f4f).
Once conclusive, we will implement the solution for Maya as well.

### CODE REVIEW

**Comment by noalak** [comment]
Next is TECHNICAL notes on the export animation only feature.

The implementation of this feature has been discussed in [this issue](https://github.com/BabylonJS/Exporters/issues/537) and [this PR](https://github.com/BabylonJS/Exporters/pull/691) and other nested threads.
 The user is responsible of node's name uniqueness (1st solution chosen).

Regarding exported data when exporting only animations:
- meshes, lights and cameras are exported as nodes (=dummies)
- the nodes exported are only the relevant ones: either animated nodes or their parent
- this mean that non-animated nodes on leaves are not exported
- non-animated intermediate nodes need to be exported as they are required by the glTF loader of babylon
- materials and other global or custom parameters are not exported

Regarding implementation:
- meshes are exported as dummies immediately to avoid a pass on useless features (mostly geometry)
- skeletons **are not** exported (a skeleton is a link between a mesh and a set of bones and will not vary between animations)
- bones **are** exported (they are the animated dummies)
- cameras and lights are first exported completely, then their transform and animations get fixed and finally they are converted back to simple nodes. The time required to completely export a camera or light is minimal anyway.

**Comment by Selmar** [comment]
I don't think it's a surprise, but since you have chosen the path of unique names, we will most likely keep our own implementation on our side and won't be using this for our pipeline.

**Comment by Drigax** [comment]
@noalak the larger the PR, the harder it is to review and test. Can we please address a single issue for each PR moving forward? Especially in cases like this where the issues aren't related at all

File: 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs [hunk 1]
```diff
@@ -1,887 +1,933 @@
...
-                }
-                File.Move(sourceFilePath, targetFilePath);
-                RaiseMessage(sourceFilePath + " -> " + targetFilePath);
-            }
-        }
-
-        private BabylonNode exportNodeRec(IIGameNode maxGameNode, BabylonScene babylonScene, IIGameScene maxGameScene)
-        {
-            BabylonNode babylonNode = null;
+﻿using Autodesk.Max;
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs]: Why are the whitespace messed up in this file?

**Comment by Drigax** [comment]
Also, the whitespace changes are blocking the auto-merge

**Comment by noalak** [comment]
Whitespace fixed.

**Comment by noalak** [comment]
We will definitely make a PR for each feature in the future.
For documentation, do you want a single PR or several ones ?

File: 3ds Max/Max2Babylon/Forms/ExporterForm.cs [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Utilities;
+using Color = System.Drawing.Color;
+using Microsoft.WindowsAPICodePack.Dialogs;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Max2Babylon
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Forms/ExporterForm.cs]: Looks like we have the same whitespace problem here as well, @noalak can your team please fix this as well? This should be good for checkin after :)

**Comment by Drigax** [comment]
@noalak , for the documentation, a single PR should be sufficient for the exporter UI changes

File: 3ds Max/Max2Babylon/Forms/ExporterForm.cs (hunk 2 above)
Comment (noalak) [reply 3ds Max/Max2Babylon/Forms/ExporterForm.cs]: Done

**Pull request merged**